src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -3892,13 +3976,27 @@ class StubGenerator: public StubCodeGenerator {
     __ align(32);
     __ bind(unrolled_loop_start);
     for (unsigned unroll_cnt=0; unroll_cnt < loop_unrolls; unroll_cnt++) {
-        // We can use a static displacement in the load since it's always a
-        // multiple of 16, which is a requirement of lxv/stxv.  This saves
-        // an addi instruction.
-        __ lxv(input->to_vsr(), unroll_cnt * 16, in);
-        //
-        // Lookup
-        //
+      // We can use a static displacement in the load since it's always a
+      // multiple of 16, which is a requirement of lxv/stxv.  This saves
+      // an addi instruction.
+      __ lxv(input->to_vsr(), unroll_cnt * 16, in);
+
+      //
+      // Lookup
+      //
+      if (PowerArchitecturePPC64 >= 10) {
+        // Use xxpermx to do a lookup of each Base64 character in the
+        // input vector and translate it to a 6-bit value + 0x80.
+        // Characters which are not valid Base64 characters will result
+        // in a zero in the corresponding byte.
+        __ align_prefix(); __ xxpermx(xlate_a, table_32_47, table_48_63, input->to_vsr(), 1);
+        __ align_prefix(); __ xxpermx(xlate_b, table_64_79, table_80_95, input->to_vsr(), 2);
+        __ xxlor(xlate_b, xlate_a, xlate_b);
+        __ align_prefix(); __ xxpermx(xlate_a, table_96_111, table_112_127, input->to_vsr(), 3);
```

Comment:
This is just a reminder comment.  align_prefix() does nothing when loop_unrolls is 1 because the label unrolled_loop_start is aligned to a 32-byte boundary and these three prefixed instructions are within 32bytes after the label, and thus, they never cross a 64-byte boundary.  Since the comment above mentions that the best unrolling factor here is 1, align_prefix() are just safe guards for the case someone try to unroll this loop.

Reply:
That's a good point. Initially I had removed the align_prefix() calls for that reason and added a comment about why they weren't needed, but it burnt me when I was testing an unroll value of 4.
I will put in a comment like you asked.

Reply:
I guess we will never change the unroll factor. Why not clean it up?

Reply:
> I guess we will never change the unroll factor. Why not clean it up?

I debated with myself back and forth on that.  My thinking was that perhaps it will make sense on later processors, like Power11.  On the other hand, it's some technical debt to have that complexity without the need.

I'll push another commit to remove it (after testing it, of course)

Reply:
I believe I have resolved this with my latest commit to this PR.  Please double check.